controllers/clusterresources/postgresqluser_controller.go

```diff
@@ -259,12 +321,72 @@ func (r *PostgreSQLUserReconciler) createUser(
 	createUserQuery := fmt.Sprintf(`CREATE USER "%s" WITH PASSWORD '%s'`, newUserName, newPassword)
 	_, err = conn.Exec(ctx, createUserQuery)
 	if err != nil {
-		return fmt.Errorf("cannot execute creation user query in postgresql, err: %w", err)
+		return fmt.Errorf("cannot execute creation user query in PostgreSQL, err: %w", err)
 	}
 
 	return nil
 }
 
+func (r *PostgreSQLUserReconciler) deleteUser(
+	ctx context.Context,
+	newUserName string,
+	defaultUserSecretNamespacedName clusterresourcesv1beta1.NamespacedName,
+) error {
+	defaultUsername, defaultPassword, clusterName, err := r.getDefaultPostgreSQLUserCreds(ctx, defaultUserSecretNamespacedName)
+	if err != nil {
+		return err
+	}
+
+	serviceName := fmt.Sprintf(models.ExposeServiceNameTemplate, clusterName)
+	host := fmt.Sprintf("%s.%s", serviceName, defaultUserSecretNamespacedName.Namespace)
+
+	dbURL := fmt.Sprintf("postgres://%s:%s@%s:%d/%s?target_session_attrs=read-write",
```

Comment:
I think it would be better to move such operations to separate method, because there are multiple places where you use it